### PR TITLE
[2.13.x] DDF-4454 Support for the Dominion Test Framework

### PR DESCRIPTION
**Pull request by paouelle**
#### What does this PR do?
This PR adds support for the dominion test framework to DDF. It defines the initial set of Dominion options which allows one to install the DDF kernel or the DDF distribution. Other test-specific options are added as well.

It fixes profile install command to properly fail the command if the profile cannot be located. This allows testing to detect failures when executing the install command via SSH.

It also modifies pom fix to allow a project to reference a feature installed by its own pom file.

#### Who is reviewing it? 

@peterhuffer 
@kcover 
@AzGoalie 
@lessarderic 

#### Select relevant component teams: 
@codice/test 

#### Ask 2 committers to review/merge the PR and tag them here.
@AzGoalie
@djblue
@clockard

#### How should this be tested?

- Successful PR build should be sufficient
- verify profile:install command when it succeeds and fails

#### Any background context you want to provide?
These changes are required to be done on the release branch of DDF since replication is currently based on that branch.

Here is a sample test class:

```
@DDFCommonOptions.ConfigureVMOptionsForTesting
@DDFCommonOptions.ConfigureDebugging
@DDFCommonOptions.ConfigurePorts
@DDFCommonOptions.ConfigureLogging
@DDFCommonOptions.PolicyFile(
  name = "some.name",
  artifact =
      @MavenUrl(
        groupId = "some.group",
        artifactId = "some.artifact",
        version = MavenUrl.AS_IN_PROJECT,
        type = "policy"
      )
)
@DDFOptions.InstallDistribution(solr = true)
@ServiceAdmin
@Feature.Start(
  repository =
      @MavenUrl(
        groupId = "some.group",
        artifactId = "some.artifact",
        version = MavenUrl.AS_IN_PROJECT,
        type = "xml",
        classifier = "features"
      ),
  name = "some.feature"
)
@RunWith(Dominion.class)
public class ITStart {
  @BeforeClass
  public static void setupClass() {
    System.out.println("*** SETUP CLASS");
  }

  @AfterClass
  public static void cleanupClass() {
    System.out.println("*** AFTER CLASS");
  }

  @Before
  public void setup() {
    System.out.println("*** SETUP");
  }

  @After
  public void cleanup() {
    System.out.println("*** AFTER");
  }

  @Test
  public void test1() throws Exception {
    System.out.println("*** TEST 1");
  }

  @Test
  public void test3() throws Exception {
    System.out.println("*** TEST 3");
  }
}
```
#### What are the relevant tickets?
For Jira:
[DDF-4454](https://codice.atlassian.net/browse/DDF-4454)

For GH Issues:
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ethantmanns** [comment]
I think we need some more context here. Are there docs anywhere I can read to learn more about "Dominion"? Should this be in the bugfix release branch?

**Comment by paouelle** [comment]
@ethantmanns This is in the release branch because the replication application will be using it and it is defined on top of the release branch of DDF. It will be merged to master later.

The Dominion Test framework is the new component/itest/blackbox framework that cane from the prototype that was done before the end of the year in the green team. It is still in development. There is currently a PR for the introduction of the framework as part of the codice-test repository (https://github.com/codice/codice-test/pull/27).

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

File: libs/dominion-commons-pax-exam/src/main/java/org/codice/ddf/dominion/commons/pax/exam/internal/CommonsOptionExtensionFactory.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.codice.ddf.dominion.commons.pax.exam.options.extensions.InstallExtension;
+import org.codice.dominion.options.Option;
+import org.codice.dominion.options.Option.Extension;
+
+/** Option factory for DDF common options that works with Dominion PaxExam driver. */
+public class CommonsOptionExtensionFactory implements Option.Factory {
+  private static final Map<Class<? extends Annotation>, Extension> EXTENSIONS;
+
+  static {
+    final Map<Class<? extends Annotation>, Option.Extension> map = new HashMap<>(8);
```
[the remark this answers is not in the document]
Comment (paouelle) [reply libs/dominion-commons-pax-exam/src/main/java/org/codice/ddf/dominion/commons/pax/exam/internal/CommonsOptionExtensionFactory.java]: Yup. Only 2 entries are added so there is no real need to set that to more.

File: distribution/ddf-dominion-pax-exam/src/main/java/org/codice/ddf/dominion/pax/exam/internal/OptionExtensionFactory.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.codice.ddf.dominion.pax.exam.options.extensions.InstallDistributionExtension;
+import org.codice.ddf.dominion.pax.exam.options.extensions.InstallKernelExtension;
+import org.codice.dominion.options.Option;
+
+/** Factory implementation for DDF's option extensions. */
+public class OptionExtensionFactory implements Option.Factory {
+  private static final Map<Class<? extends Annotation>, Option.Extension> EXTENSIONS;
+
+  static {
+    final Map<Class<? extends Annotation>, Option.Extension> map = new HashMap<>(6);
```
[the remark this answers is not in the document]
Comment (paouelle) [reply distribution/ddf-dominion-pax-exam/src/main/java/org/codice/ddf/dominion/pax/exam/internal/OptionExtensionFactory.java]: Yup. Only 2 entries are added so there is no real need to set that to more.

File: distribution/ddf-dominion-pax-exam/pom.xml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>ddf</groupId>
+        <artifactId>distribution</artifactId>
+        <version>2.13.6-SNAPSHOT</version>
+    </parent>
+
+    <groupId>org.codice.ddf</groupId>
+    <artifactId>ddf-dominion-pax-exam</artifactId>
+    <name>DDF :: Distribution :: DDF :: Dominion :: Pax Exam Extensions</name>
```
[the remark this answers is not in the document]
Comment (paouelle) [reply distribution/ddf-dominion-pax-exam/pom.xml]: removed

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now
looks like it built the wrong commit

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now
rebuilt locally with a clean .m2 and no issues were encoutered

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now
after reformatting a file

**Comment by paouelle** [comment]
build now
yarn download error

**Comment by kcover** [comment]
Successful hero! 🎉 Installing with an unknown profile gave me 
`Profile: bob not found
Error executing command: Profile: bob not found`
and installing the standard profile completed successfully

**Pull request merged**